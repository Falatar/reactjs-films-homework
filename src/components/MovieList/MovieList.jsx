import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import style from './MovieList.scss';
import FilmBlock from '../FilmBlock';

class MovieList extends Component {
  componentDidMount() {
    const { getFilmList } = this.props;
    getFilmList();
  }

  render() {
    const {
      filmList,
      searchResult,
    } = this.props;
    if (!searchResult) {
      return (
        <div className={classNames(style.list, style.empty_list)}>
          <span className={style.notice}>There are 0 results for your request</span>
        </div>
      );
    }
    if (filmList.length) {
      return (
        <div className={style.list}>
          {filmList.map((item) => (
            <FilmBlock
              id={item.id}
              name={item.title}
              imgURL={item.backdrop_path}
              rating={item.vote_average}
              tagList={item.genre_str}
              overview={item.overview}
              key={item.id}
            />
          ))}
        </div>
      );
    }
    return (
      <div className={classNames(style.list, style.empty_list)}>
        <span className={style.notice}>Loading...</span>
      </div>
    );
  }
}

MovieList.defaultProps = {
  filmList: [],
  searchResult: true,
  getFilmList: () => {},
};

MovieList.propTypes = {
  filmList: PropTypes.objectOf(PropTypes.any),
  searchResult: PropTypes.bool,
  getFilmList: PropTypes.func,
};

export default MovieList;
