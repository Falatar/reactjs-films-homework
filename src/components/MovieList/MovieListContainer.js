import { connect } from 'react-redux';
import { loadActualFilms, loadGenres } from '../../modules/filmListModule/filmListModuleActions';
import { getMoviesInfo, getSearchResult } from '../../modules/filmListModule/filmListModuleSelector';
import MovieList from './MovieList';

export const mapStateToProps = (store) => ({
  filmList: getMoviesInfo(store),
  searchResult: getSearchResult(store),
});

export const mapDispatchToProps = (dispatch) => ({
  getFilmList: () => dispatch(loadActualFilms()),
  loadGenreList: () => dispatch(loadGenres()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MovieList);
