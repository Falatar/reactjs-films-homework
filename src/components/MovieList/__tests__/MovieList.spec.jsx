import React from 'react';
import renderer from 'react-test-renderer';
import MovieList from '../MovieListContainer';

describe('MoviewList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MovieList />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
