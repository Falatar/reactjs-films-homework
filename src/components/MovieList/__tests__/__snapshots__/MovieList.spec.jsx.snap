// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders correctly without input 1`] = `
<div
  className="list empty_list"
>
  <span
    className="notice"
  >
    Loading...
  </span>
</div>
`;

exports[`MovieList renders correctly without input and results 1`] = `
<div
  className="list empty_list"
>
  <span
    className="notice"
  >
    There are 0 results for your request
  </span>
</div>
`;

exports[`MovieList renders correctly without results 1`] = `
<div
  className="list empty_list"
>
  <span
    className="notice"
  >
    There are 0 results for your request
  </span>
</div>
`;
